### PR TITLE
api: add focal parameters to the radial apis

### PR DESCRIPTION
**Pull request by mgrudzinska**
@Issue: https://github.com/thorvg/thorvg/issues/2860

### CODE REVIEW

File: inc/thorvg.h [hunk 1]
```diff
@@ -767,31 +767,40 @@ class TVG_API RadialGradient final : public Fill
...
+     * @param[in] fr The radius of the start circle.
      *
-     * @retval Result::InvalidArguments in case the @p radius value is zero or less.
+     * @retval Result::InvalidArguments in case the radius @p r or @p fr value is negative.
      *
-     * @note In case the @p radius is zero, an object is filled with a single color using the last color specified in the colorStops().
+     * @note In case the radius @p r is zero, an object is filled with a single color using the last color specified in the colorStops().
+     * @note If you choose not to specify a focal point, ensure that @p fx and @p fy are the same as @p cx and @p cy, and set @p fr to 0.
      */
-    Result radial(float cx, float cy, float radius) noexcept;
```
Comment (hermet) [inline inc/thorvg.h:782]: We can likely specify the usage of the focal:
```
By adjusting the position and size of the focal point, you can create various visual effects, such as focusing the gradient closer to one edge or generating more complex shading patterns. If you don't need a focal effect, set the focal point values to be the same as the center with a zero radius.
```

File: test/testFill.cpp [hunk 2]
```diff
@@ -151,24 +151,31 @@ TEST_CASE("Radial Filling", "[tvgFill]")
...
-    REQUIRE(fill->radial(0, 0, -1) == Result::InvalidArguments);
-    REQUIRE(fill->radial(nullptr, nullptr, nullptr) == Result::Success);
-    REQUIRE(fill->radial(100, 120, 50) == Result::Success);
+    REQUIRE(fill->radial(0, 0, -1, 0, 0, 0) == Result::InvalidArguments);
+    REQUIRE(fill->radial(0, 0, 0, 0, 0, -1) == Result::InvalidArguments);
+    REQUIRE(fill->radial(nullptr, nullptr, nullptr, nullptr, nullptr, nullptr) == Result::Success);
+    REQUIRE(fill->radial(100, 120, 50, 10, 20, 5) == Result::Success);
 
-    REQUIRE(fill->radial(&cx, nullptr, &radius) == Result::Success);
+    REQUIRE(fill->radial(&cx, nullptr, &r, nullptr, nullptr, nullptr) == Result::Success);
```
Comment (hermet) [inline test/testFill.cpp:161]: let's remove the nullptr passes for focal.

Comment (mgrudzinska) [reply test/testFill.cpp:161]: please check now, removed also in l.158

File: inc/thorvg.h (hunk 1 above)
Comment (mgrudzinska) [reply inc/thorvg.h:782]: please check now, changed in thorvg.h and capi version as well.

**Pull request merged**